public/pages/AnomalyCharts/containers/AnomalyDetailsChart.tsx

```diff
@@ -0,0 +1,388 @@
+        !props.isHCDetector
+      ) {
+        getMonitorAlerts(
+          props.monitor.id,
+          props.dateRange.startDate,
+          props.dateRange.endDate
+        );
+      }
+    }, [props.monitor, props.dateRange.startDate]);
+
+    const anomalyChartTimeFormatter = niceTimeFormatter([
+      zoomRange.startDate,
+      zoomRange.endDate,
+    ]);
+
+    const handleDateRangeChange = (startDate: number, endDate: number) => {
+      props.onDateRangeChange(startDate, endDate);
+      handleZoomRangeChange(startDate, endDate);
+    };
+
+    const showLoader = useDelayedLoader(props.isLoading || isLoadingAlerts);
+
+    return (
+      <React.Fragment>
+        <EuiFlexGroup style={{ padding: '20px' }}>
+          <EuiFlexItem>
+            <EuiStat
+              title={
+                props.isLoading || isLoadingAlerts
+                  ? '-'
+                  : anomalySummary.anomalyOccurrence
+              }
+              description={
+                props.showAlerts
+                  ? 'Anomaly occurrences'
+                  : 'Sample anomaly occurrences'
+              }
+              titleSize="s"
+            />
+          </EuiFlexItem>
+          <EuiFlexItem>
+            <AnomalyStatWithTooltip
+              isLoading={props.isLoading || isLoadingAlerts}
+              minValue={anomalySummary.minAnomalyGrade}
+              maxValue={anomalySummary.maxAnomalyGrade}
+              description={
+                props.showAlerts ? 'Anomaly grade' : 'Sample anomaly grade'
+              }
+              tooltip="Indicates to what extent this data point is anomalous. The scale ranges from 0 to 1."
+            />
+          </EuiFlexItem>
+          <EuiFlexItem>
+            <AnomalyStatWithTooltip
+              isLoading={props.isLoading || isLoadingAlerts}
+              minValue={anomalySummary.minConfidence}
+              maxValue={anomalySummary.maxConfidence}
+              description={
+                props.showAlerts ? 'Confidence' : 'Sample confidence'
+              }
+              tooltip="Indicates the level of confidence in the anomaly result."
+            />
+          </EuiFlexItem>
+          <EuiFlexItem>
+            <EuiStat
+              title={
+                props.isLoading || isLoadingAlerts
+                  ? ''
+                  : anomalySummary.lastAnomalyOccurrence
+              }
+              description={
+                props.showAlerts
+                  ? 'Last anomaly occurrence'
+                  : 'Last sample anomaly occurrence'
+              }
+              titleSize="s"
+            />
+          </EuiFlexItem>
+          {props.showAlerts && !props.isHCDetector ? (
+            <EuiFlexItem>
+              <AlertsStat
+                monitor={props.monitor}
+                showAlertsFlyout={() => setShowAlertsFlyout(true)}
+                totalAlerts={totalAlerts}
+                isLoading={props.isLoading}
```

Comment:
should `isLoading` here also include `isLoadingAlerts`? For example, change to `isLoading={props.isLoading || isLoadingAlerts}

Reply:
Actually, looks like all places where `props.isLoading` is called, `isLoadingAlerts` is called as well (except for this one case above). Could maybe add a variable at the top that's combining these loading states into one? For example `const isLoading = props.isLoading || isLoadingAlerts`?

Reply:
good point. I will change to `const isLoading = props.isLoading || isLoadingAlerts` for clean code.